pydantic/_internal/_core_utils.py

```diff
@@ -340,6 +340,15 @@ def handle_typed_dict_schema(self, schema: core_schema.TypedDictSchema) -> CoreS
         schema['fields'] = replaced_fields
         return schema
 
+    def handle_dataclass_args_schema(self, schema: core_schema.DataclassArgsSchema) -> CoreSchema:
+        replaced_fields: list[core_schema.DataclassField] = []
+        for field in schema['fields']:
+            replaced_field = field.copy()
+            replaced_field['schema'] = self._walk(field['schema'])
+            replaced_fields.append(replaced_field)
+        schema['fields'] = replaced_fields
+        return schema
```

Comment:
This class recurses over a core schema applying a transform to nested item of type `CoreSchema`.

I can add a docstring here, but presumably I should add a docstring to the class and/or every method within the class. I wasn't sure whether this would be long lived when I first made it which is why I didn't doc anything, but at this point I think it has proven its utility enough that I would strongly argue for it to stay (though I think it may be better to put it in `pydantic_core`, as noted at the top of the module).